File: Backend/Source/src/Models/Order.cs
using System.Collections.Generic;

namespace src.Models;

public class Order
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public int? AddressId { get; set; }
    public string OrderNumber { get; set; } = string.Empty;
    public decimal Subtotal { get; set; }
    public decimal ShippingFee { get; set; }
    public decimal TotalAmount { get; set; }
    public string Status { get; set; } = "Pending";
    public string PaymentStatus { get; set; } = "Unpaid";
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User? User { get; set; }
    public Address? Address { get; set; }
    public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
}

File: Frontend/my-app/src/types/order.ts
/**
 * Order DTOs mirrored from the .NET backend
 * (Backend/Source/src/DTOs/{OrderDto,CheckoutDto}.cs).
 */

/** POST /api/orders/{userId} — request body. */
export interface CheckoutRequest {
  firstName?: string;
  lastName?: string;
  email?: string;
  address?: string;
  city?: string;
  state?: string;
  postalCode?: string;
  country?: string;
  recipientName?: string;
  phoneNumber?: string;
  addressId?: number;
  paymentMethod?: string;
  payment?: string;
}

/** POST /api/orders/{userId} — response `data` payload. */
export interface OrderDto {
  id: number;
  orderNumber: string;
  status: string;
  paymentStatus: string;
  subtotal: number;
  shippingFee: number;
  totalAmount: number;
  createdAt: string;
  address: AddressDto | null;
  items: OrderItemDto[];
}

export interface AddressDto {
  recipientName: string;
  phoneNumber: string;
  street: string;
  city: string;
  state: string;
  postalCode: string;
  country: string;
  email?: string | null;
}

export type OrderStatus = 'Pending' | 'Processing' | 'Shipped' | 'Delivered' | 'Cancelled'

export interface OrderItemDto {
  productId: number;
  productVariantId: number;
  productName: string;
  sizeName: string;
  color: string | null;
  imageUrl: string | null;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}
